examples/GettingFeaturesFromDatabase/Program.cs

```diff
@@ -0,0 +1,23 @@
+using GettingFeaturesFromDatabase;
+using GettingFeaturesFromDatabase.Database;
+using Microsoft.FeatureManagement;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddDatabase();
+builder.Services.AddFeatureService();
+
+/* Add feature management */
+builder.Services.AddScoped<IFeatureDefinitionProvider, CustomFeatureDefinitionProvider>().AddScopedFeatureManagement();
```

Comment:
Hi, @fffffatah Maybe this is a stupid question. May I know the reason why CustomFeatureDefinitionProvider is registered as scoped here?

Reply:
Hi @zhiyuanliang-ms 👋, CustomFeatureDefinitionProvider is scoped here because the life-time of DbContext and the service which is used to read the features from DB are scoped. If I'm not wrong, setting the life time of DbContext as singleton might lead to issues when many concurrent calls are made. Although, DbContext can be registered as singleton and the Feature set can be marked AsNoTracking while querying, but it felt a bit risky.

Reply:
Thank you!

Reply:
Thank you!